clippy_lints/src/multiple_crate_versions.rs

```diff
@@ -0,0 +1,72 @@
+//! lint on multiple versions of a crate being used
+
+use rustc::lint::*;
+use syntax::ast::*;
+
+use cargo_metadata;
+use itertools::Itertools;
+
+/// **What it does:** Checks to see if multiple versions of a crate are being
+/// used.
+///
+/// **Why is this bad?** This bloats the size of targets, and can lead to
+/// confusing error messages when structs or traits are used interchangeably
+/// between different versions of a crate.
+///
+/// **Known problems:** Because this can be caused purely by the dependencies
+/// themselves, it's not always possible to fix this issue.
+///
+/// **Example:**
+/// ```toml
+/// # This will pull in both winapi v0.3.4 and v0.2.8, triggering a warning.
+/// [dependencies]
+/// ctrlc = "3.1.0"
+/// ansi_term = "0.11.0"
+/// ```
+declare_clippy_lint! {
+    pub MULTIPLE_CRATE_VERSIONS,
+    nursery,
```

Comment:
We should create a new lint group `cargo` in the `declare_clippy_lint` macro and use that instead of `nursery`. Otherwise the lint update Python script will move the lint back to the nursery